src/proto/h2/ping.rs

```diff
@@ -282,12 +310,11 @@ impl Ponger {
             return Poll::Pending;
         }
 
-        let (bytes, rtt) = match locked.ping_pong.poll_pong(cx) {
+        match locked.ping_pong.poll_pong(cx) {
             Poll::Ready(Ok(_pong)) => {
-                let rtt = locked
+                let rtt = now - locked
                     .ping_sent_at
-                    .expect("pong received implies ping_sent_at")
-                    .elapsed();
+                    .expect("pong received implies ping_sent_at");
```

Comment:
Nit: binding `locked.ping_sent_at.expect(…)` to its own variable before computing `rtt` would probably help readability here, the line feeds mixed with the subtraction feels hard to follow to me.